commit a72b46c3849cdb05993015991bde548ab8b6d7ac
Author: Ying Huang <huang.ying.caritas@gmail.com>
Date:   Fri Apr 24 10:45:17 2009 +0800

    PCI: Add pci_bus_set_ops
    
    pci_bus_set_ops changes pci_ops associated with a pci_bus. This can be
    used by debug tools such as PCIE AER error injection to fake some PCI
    configuration registers.
    
    Acked-by: Kenji Kaneshige <kaneshige.kenji@jp.fujitsu.com>
    Signed-off-by: Huang Ying <ying.huang@intel.com>
    Signed-off-by: Jesse Barnes <jbarnes@virtuousgeek.org>

diff --git a/include/linux/pci.h b/include/linux/pci.h
index ec03b90d3510..ea2a153a9126 100644
--- a/include/linux/pci.h
+++ b/include/linux/pci.h
@@ -637,6 +637,7 @@ int pci_bus_write_config_word(struct pci_bus *bus, unsigned int devfn,
 			      int where, u16 val);
 int pci_bus_write_config_dword(struct pci_bus *bus, unsigned int devfn,
 			       int where, u32 val);
+struct pci_ops *pci_bus_set_ops(struct pci_bus *bus, struct pci_ops *ops);
 
 static inline int pci_read_config_byte(struct pci_dev *dev, int where, u8 *val)
 {